hal/src/peripherals/calibration/d5x.rs

```diff
@@ -6,15 +6,35 @@ use core::ptr;
 // "The NVM Software Calibration Area can be read at address 0x00800080."
 const ADDR: u32 = 0x00800080;
 
+// Calculations for this from the datasheet values:
+// Example: Bit position: 41:37
+//
+// addr_offset = int(37/8) = 4
+// bit_shift = 37 % 8 = 5
+// Mask length = 41-37+1 = 5 bits (0b11111)
 fn cal(addr_offset: u32, bit_shift: u32, bit_mask: u32) -> u32 {
     unsafe {
         let addr: *const u32 = (ADDR + addr_offset) as *const _;
-        let value = ptr::read(addr);
+        let value = ptr::read_unaligned(addr);
 
         (value >> bit_shift) & bit_mask
     }
 }
 
+// Needed for temperature calibration values stored in NVM
+fn parts_to_f32(int: u32, dec: u32) -> f32 {
+    let mut dec = dec as f32;
+
+    if dec < 10.0 {
```

Comment:
This if/else seems suspect to me, it interprets both the scale and the value from the `dec` parameter.  The datasheet is really poor in this area, but my interpretation is that dec is a BCD coded field, so the method should simply be (probably not a function):

(int * 10 + dec) as f32 / 10.0

Reply:
I'm playing with the CircuitPython implementation by comparison, [here's how they do it](https://github.com/adafruit/circuitpython/blob/002881826a04669abaaa4f2d5220379f786cb159/ports/atmel-samd/common-hal/microcontroller/Processor.c#L94-L108) - looks to be BCD (dec is the value in 1/10 degree C).

Reply:
Sorry for all the reply-to-self on this, it's taking me a while to get caught up on the docs.  The D21 datasheet talks a bit more about the format in 37.11.8.2.1 Temperature Log Row, it's BCD so I'll PR a small change